backend/ecs_tasks/delete_files/parquet_handler.py

```diff
@@ -114,12 +112,16 @@ def cast_column_values(column, schema):
     """
     if column["Type"] == "Simple":
         if is_column_type_decimal(schema, column["Column"]):
-            column["MatchIds"] = [Decimal(m) for m in column["MatchIds"]]
+            column["MatchIds"] = set([Decimal(m) for m in column["MatchIds"]])
```

Comment:
```suggestion
            column["MatchIds"] = set(Decimal(m) for m in column["MatchIds"])
```

You can skip creating an intermediate list by omitting the braces.